softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/RouterFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.softgroup.common.router.api;
+
+import com.softgroup.common.protocol.Request;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 25.02.17.
+ */
+
+public class RouterFactory<T extends CommonRouterHandler>  {
```

Comment:
I wanted to do abstract parent for both , and overriding getRouteKey method, but  I don`t know how to do it correctly (some problem with generics)